susemanager-utils/susemanager-sls/salt/bootstrap/init.sls

```diff
@@ -127,16 +129,39 @@ bootstrap_repo:
       - ([ {{ bootstrap_repo_exists }} = "True" ])
 {%- endif %}
 
+{%- if grains['os_family'] == 'RedHat' %}
+trust_res_gpg_key:
+  cmd.run:
+    - name: rpm --import https://{{ salt['pillar.get']('mgr_server') }}/pub/{{ salt['pillar.get']('gpgkeys:res:file') }}
+    - unless: rpm -q {{ salt['pillar.get']('gpgkeys:res:name') }}
+    - runas: root
+{%- elif grains['os_family'] == 'Debian' %}
+install_gnupg_debian:
+  pkg.latest:
+    - pkgs:
+      - gnupg
+{%- endif %}
+
 {% include 'channels/gpg-keys.sls' %}
 
+{%- set salt_minion_name = 'salt-minion' %}
+{%- set salt_config_dir = '/etc/salt' %}
+{% set venv_available_request = salt['http.query'](bootstrap_repo_url + 'venv-enabled-' + grains['osarch'] + '.txt', status=True, verify_ssl=False) %}
+{# Prefer venv-salt-minion if available and not disabled #}
+{%- set use_venv_salt = (0 < venv_available_request.get('status', 404) < 300) and not salt['pillar.get']('mgr_avoid_venv_salt_minion') %}
+{%- if use_venv_salt %}
+{%- set salt_minion_name = 'venv-salt-minion' %}
+{%- set salt_config_dir = '/etc/opt/venv-salt-minion' %}
+{%- endif -%}
```

Comment:
I wonder if it is possible to move this to the top of the file.
As this is living in the "general" section without limit to a special OS it would make sense to have all the "set" rules at the top.

Reply:
Unfortunately `bootstrap_repo_url` is calculated far from the top of the sls file, in this case only first 2 sets can e moved to the top. Not so good idea to move just to lines far from the rest logic

Reply:
ahh, ok.